cirq/google/optimizers/convert_to_sycamore_gates.py

```diff
@@ -134,68 +141,124 @@ def optimization_at(self, circuit, index, op):
                                                  new_operations=converted,
                                                  clear_qubits=op.qubits)
 
-    def known_two_q_operations_to_sycamore_operations(self, qubit_a: ops.Qid,
-                                                      qubit_b: ops.Qid,
-                                                      op: ops.GateOperation
-                                                     ) -> ops.OP_TREE:
-        """
-        Synthesize a known gate operation to a sycamore operation
 
-        This function dispatches based on gate type
+def known_two_q_operations_to_sycamore_operations(
+        qubit_a: ops.Qid,
+        qubit_b: ops.Qid,
+        op: ops.GateOperation,
+        tabulation: Optional[GateTabulation] = None) -> ops.OP_TREE:
+    """
+    Synthesize a known gate operation to a sycamore operation
+
+    This function dispatches based on gate type
+
+    Args:
+        qubit_a: first qubit of GateOperation
+        qubit_b: second qubit of GateOperation
+        op: operation to decompose
+        tabulation: A tabulation for the Sycamore gate to use for
+            decomposing gates.
+    Returns:
+        New operations iterable object
+    """
+    gate = op.gate
+    if isinstance(gate, ops.CNotPowGate):
+        return [
+            ops.Y(qubit_b)**-0.5,
+            cphase(
+                cast(ops.CNotPowGate, gate).exponent * np.pi, qubit_a, qubit_b),
+            ops.Y(qubit_b)**0.5,
+        ]
+    elif isinstance(gate, ops.CZPowGate):
+        gate = cast(ops.CZPowGate, gate)
+        if math.isclose(gate.exponent, 1.0):  # check if CZ or CPHASE
+            return decompose_cz_into_syc(qubit_a, qubit_b)
+        else:
+            # because CZPowGate == diag([1, 1, 1, e^{i pi phi}])
+            return cphase(gate.exponent * np.pi, qubit_a, qubit_b)
+    elif isinstance(gate, ops.SwapPowGate) and math.isclose(
+            cast(ops.SwapPowGate, gate).exponent, 1.0):
+        return decompose_swap_into_syc(qubit_a, qubit_b)
+    elif isinstance(gate, ops.ISwapPowGate) and math.isclose(
+            cast(ops.ISwapPowGate, gate).exponent, 1.0):
+        return decompose_iswap_into_syc(qubit_a, qubit_b)
+    elif isinstance(gate, ops.ZZPowGate):
+        return rzz(cast(ops.ZZPowGate, gate).exponent * np.pi / 2, *op.qubits)
+    elif isinstance(gate, ops.MatrixGate) and len(op.qubits) == 2:
+        if tabulation:
+            return decompose_arbitrary_into_syc_tabulation_(
+                qubit_a, qubit_b, op, tabulation)
+        else:
+            return decompose_arbitrary_into_syc_analytic_(qubit_a, qubit_b, op)
+    else:
+        raise ValueError("Unrecognized gate: {!r}".format(op))
 
-        Args:
-            qubit_a: first qubit of GateOperation
-            qubit_b: second qubit of GateOperation
+
+def decompose_arbitrary_into_syc_tabulation_(qubit_a: ops.Qid, qubit_b: ops.Qid,
```

Comment:
why do these methods have a trailing `_`? afaik this isn't a convention in python

Reply:
I plead the fifth.